starknet-foundry/crates/forge/src/cheatcodes_hint_processor.rs

```diff
@@ -747,3 +713,65 @@ mod test {
         }
     }
 }
+
+fn convert_to_blockifier_felt(val: &Felt252) -> blockifier_Felt252 {
+    let v = val.to_bigint();
+    blockifier_Felt252::from(v)
+}
+
+fn convert_from_blockifier_felt(val: &blockifier_Felt252) -> Felt252 {
+    let v = val.to_bigint();
+    Felt252::from(v)
+}
+
+#[cfg(test)]
+mod test_felt_conversions {
+    use super::convert_from_blockifier_felt;
+    use super::convert_to_blockifier_felt;
+    use cairo_felt::{
+        Felt252 as cairo_Felt252, FIELD_HIGH as cairo_FIELD_HIGH, FIELD_LOW as cairo_FIELD_LOW,
+    };
+    use cairo_felt_blockifier::Felt252 as blockifier_Felt252;
+
+    use num_traits::ToPrimitive;
+
+    // Does not re-export the consts
+    const BLOCKIFIER_FIELD_HIGH: u128 = (1 << 123) + (17 << 64);
+    const BLOCKIFIER_FIELD_LOW: u128 = 1;
+
+    #[test]
+    fn test_conversion_to_blockifier_felt() {
+        for value in [
+            BLOCKIFIER_FIELD_LOW,
+            200_u128,
+            400_000_000_u128,
+            9_999_999_999_999_u128,
+            BLOCKIFIER_FIELD_HIGH,
+        ] {
+            let from = cairo_Felt252::from(value);
+            let to = convert_to_blockifier_felt(&from);
+            assert_eq!(
+                to.to_u128().expect("Downcast failure"),
+                from.to_u128().expect("Downcast failure")
+            );
+        }
+    }
```

Comment:
Can't we use some hardcoded expected values? Converting to and from and comparing the values seems like it could not detect some issues.

Reply:
It is about checking numerical correctness of the conversion, so it seems alright to me.

Reply:
Agree with @Arcticae here

Reply:
Not a fan of this approach